File: redsheet-portal/src/app/database-models/v-project-game-detail-record.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vProjectGameDetailRecord {
    constructor(vProjectGameDetailRecord?: vProjectGameDetailRecord )  {
        let properties = [ "currentGameId", "notes", "projectGameDetailId", "projectModuleId", "theirCurrentGameId", "theirEventGameId",];
        for (let property of properties)
            if (vProjectGameDetailRecord && vProjectGameDetailRecord[property])
                this[property] = vProjectGameDetailRecord[property];
    }
 
    @range(1,2147483647)
	currentGameId : number =   undefined;
 
    @required()
    @maxLength(1000)
	notes : string =   undefined;
 
	projectGameDetailId : number =   0 ;
 
    @range(1,2147483647)
	projectModuleId : number =   undefined;
 
    @range(1,2147483647)
	theirCurrentGameId : number =   undefined;
 
    @range(1,2147483647)
	theirEventGameId : number =   undefined;


}

File: RedSheet.DbEntities/Models/vProjectGameDetailRecord.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vProjectGameDetailRecords", Schema = "dbo")]
	[Serializable]
	public partial class vProjectGameDetailRecord  
	{
 
	[Range(1, int.MaxValue)]
	public int CurrentGameId { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string Notes { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int ProjectGameDetailId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ProjectModuleId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int TheirCurrentGameId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int TheirEventGameId { get; set; } 

	public vProjectGameDetailRecord()
	{
    }

    }
}

